caproto/_log.py

```diff
@@ -143,10 +156,120 @@ def color_logs(color):
+    ----------
+    names : string or list of string
+        PV name or PV name list which will be filtered in.
+
+    Returns
+    -------
+    Bool: True or False
+        True if message is not PV related which doesn't have 'pv' as key in extra
+        True if 'pv' as key exists and pv name exists in Filter list.
+        False if message is PV related but pv name isn't in Filter list.
+    '''
+    def __init__(self, names, level='NOTSET', exclusive=False):
+        self.names = names
+        self.levelno = validate_level(level)
+        self.exclusive = exclusive
+
+    def filter(self, record):
+        if hasattr(record, 'pv'):
+            for i in self.names:
+                if fnmatch.fnmatch(record.pv, i) and record.levelno >= self.levelno:
+                    return True
+            return False
+        else:
+            return not self.exclusive
+
+
+class AddressFilter(logging.Filter):
+
+    def __init__(self, addresses_list, level='NOTSET', exclusive=False):
+        self.addresses_list = []
+        self.hosts_list = []
+        for address in addresses_list:
+            if isinstance(address, str):
+                if ':' in address:
+                    address_temp = address.split(':')
+                    self.addresses_list.append((address_temp[0], int(address[1])))
+                else:
+                    self.hosts_list.append(address)
+            elif isinstance(address, tuple):
+                if len(address) == 2:
+                    self.addresses_list.append(address)
+                else:
+                    raise ValueError('The target addresses should be a list of string, \'XX.XX.XX.XX:YYYY\' or tuple, (\'XX.XX.XX.XX\', YYYY)')
+            else:
+                raise ValueError('The target addresses should be a list of string, \'XX.XX.XX.XX:YYYY\' or tuple, (\'XX.XX.XX.XX\', YYYY)')
+
+        self.levelno = validate_level(level)
+        self.exclusive = exclusive
+
+    def filter(self, record):
+        if hasattr(record, 'our_address') or hasattr(record, 'their_address'):
+            if record.levelno >= self.levelno:
+                return record.our_address in self.addresses_list \
+                        or record.our_address[0] in self.hosts_list \
+                        or record.their_address in self.addresses_list \
+                        or record.their_address[0] in self.hosts_list
+            else:
+                return False
+        else:
+            return not self.exclusive
+
+
+class RoleFilter(logging.Filter):
+    def __init__(self, role, level='NOTSET', exclusive=False):
+        self.roles = role
+        self.levelno = validate_level(level)
+        self.exclusive = exclusive
+
+    def filter(self, record):
+        if hasattr(record, 'role'):
+            return record.role is self.role and record.levelno >= self.levelno
+        else:
+            return not exclusive
+
+
+def set_handler(file=sys.stdout, datefmt='%H:%M:%S', level='INFO', color=True):
```

Comment:
I think this should default to 'WARNING', as it effectively does currently. Showing 'INFO' is too verbose for a default. Also, new arguments must be added at the _end_ of the signature, not inserted in the middle, in order to avoid breaking any existing user code that might call this arguments positionally.